src/app-layer-smtp.c

```diff
@@ -1302,12 +1306,13 @@ static AppLayerResult SMTPParse(uint8_t direction, Flow *f, SMTPState *state,
         while (res.status == 0) {
             if (SMTPProcessReply(state, f, pstate, thread_data) == -1)
                 SCReturnStruct(APP_LAYER_ERROR);
-            res = SMTPGetLine(state);
+            res = SMTPGetLine(state, smtp_input);
         }
         if (res.status == 1)
             return res;
     }
 
+    SMTPInputFree(smtp_input);
```

Comment:
I wonder if this would leak memory in error and incomplete states.. Gotta check.